commerce-cloud-code/app_yourshophere/cartridge/controllers/Cart.js

```diff
@@ -3,15 +3,29 @@
 const server = require('server');
 
 server.use('Show', (req, res, next) => {
-    res.page('cart', {});
+    const BasketMgr = require('dw/order/BasketMgr');
+    const HookMgr = require('dw/system/HookMgr');
+    const Transaction = require('dw/system/Transaction');
+    const ShippingMgr = require('dw/order/ShippingMgr');
+    const basket = BasketMgr.getCurrentOrNewBasket();
+
+    Transaction.wrap(() => {
```

Comment:
Ideally we do this on first add to cart - so minicart has the preview of shipping costs. But good to reaffirm this here.

Reply:
Maybe in calculate cart?
`if(lineitems.lenght > 0 && shippingMethod === null) // set default`

Reply:
ok works for me